src/realm/object-store/impl/list_notifier.cpp

```diff
@@ -49,8 +49,9 @@ void ListNotifier::attach(CollectionBase const& src)
     auto& tr = transaction();
     try {
         auto obj = tr.get_table(src.get_table()->get_key())->get_object(src.get_owner_key());
-        // FIXME: Use path for list when supporting notifications for nested collections
-        m_list = obj.get_collection_ptr(src.get_col_key());
+        auto path = src.get_short_path();
+        auto full = src.get_path().path_from_top;
+        m_list = std::dynamic_pointer_cast<CollectionBase>(obj.get_collection_ptr(path));
```

Comment:
Why is this a dynamic cast? Will the value ever not be a CollectionBase?

Reply:
`static_cast_ptr` instead of `dynamic_cast_ptr` added.